core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -2035,10 +2035,11 @@ private RecordCursorFactory generateTableQuery(
                     intervalHitsOnlyOnePartition = false;
                 }
 
-                if (intrinsicModel.keyColumn != null) {
+                if (intrinsicModel.keyColumn != null || intrinsicModel.keyExcludedColumn != null) {
                     // existence of column would have been already validated
-                    final int keyColumnIndex = reader.getMetadata().getColumnIndexQuiet(intrinsicModel.keyColumn);
+                    final int keyColumnIndex = intrinsicModel.keyColumn != null ? reader.getMetadata().getColumnIndexQuiet(intrinsicModel.keyColumn) : reader.getMetadata().getColumnIndexQuiet(intrinsicModel.keyExcludedColumn);
```

Comment:
Perhaps where clause parser should determine single key column?

I could not find tests that exercise decision making around having both '=' and '!='

Reply:
Vlad, as discussed, I've gone back to one column